sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -0,0 +1,575 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/az_mqtt5.h>
+#include <azure/core/az_mqtt5_rpc_server.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_log_internal.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "mqtt_protocol.h"
```

Comment:
We will need to implement our own topic comparison.
Also, in the future we'll need to have a more complicated topic / alias manager to optimize the data sent over the wire.